services/storage/src/simcore_service_storage/simcore_s3_dsm_utils.py

```diff
@@ -119,3 +119,20 @@ async def _get_fmd(
     directory_file_id_fmd = await _get_fmd(conn, directory_file_id)
 
     return directory_file_id if directory_file_id_fmd else None
+
+
+async def find_enclosing_file(
+    conn: SAConnection, user_id: UserID, file_id: SimcoreS3FileID
+) -> FileMetaDataAtDB | None:
+    kwnon_files = {
+        Path(known_file.file_id): known_file
+        for known_file in await db_file_meta_data.list_fmds(conn, user_id=user_id)
+    }
+    current_path = Path(file_id)
+    while current_path and current_path != Path(current_path).parent:
+        if current_path in kwnon_files:
```

Comment:
THOUGHT: This is the perfect example to use `conn.stream` when we upgrade `db_file_meta_data.list_fmds`.  This way we can stream the numbers until finding a match ...

or even better create a request that finds the match in the database?

Reply:
Obsolete, approach changed.